iron-dropdown-scroll-manager.html

```diff
@@ -33,39 +35,56 @@
 
 
       /**
-       * Returns true if the provided element is "scroll locked," which is to
-       * say that it cannot be scrolled via pointer or keyboard interactions.
+       * Returns true if the element causing the scroll is "scroll locked," which is to
+       * say that it cannot be scrolled via pointer or keyboard interactions, or
+       * if already at its scroll boundaries.
        *
-       * @param {HTMLElement} element An HTML element instance which may or may
-       * not be scroll locked.
+       * @param {Event} event The scroll event
        */
-      elementIsScrollLocked: function(element) {
-        var currentLockingElement = this.currentLockingElement;
-
-        if (currentLockingElement === undefined)
+      shouldPreventScrolling: function(event) {
+        // Avoid expensive checks if the event is not one of the observed keys.
+        if (event.type === 'keydown' && !this._isScrollingKeypress(event)) {
           return false;
-
-        var scrollLocked;
-
-        if (this._hasCachedLockedElement(element)) {
-          return true;
         }
 
-        if (this._hasCachedUnlockedElement(element)) {
+        var currentLockingElement = this.currentLockingElement;
+        if (currentLockingElement === undefined) {
           return false;
         }
 
-        scrollLocked = !!currentLockingElement &&
-          currentLockingElement !== element &&
-          !this._composedTreeContains(currentLockingElement, element);
-
-        if (scrollLocked) {
-          this._lockedElementCache.push(element);
-        } else {
-          this._unlockedElementCache.push(element);
+        var xDelta = event.wheelDeltaX;
```

Comment:
wheelDelta[XY] seems to be webkit only. Ctrl+F wheelDeltaX in the following:

https://developer.mozilla.org/en-US/docs/Web/Events/wheel